front-end/src/screens/Settings/setCredentials.tsx

```diff
@@ -0,0 +1,260 @@
+		setIsAccountLoading(false);
+		return;
+	};
+
+	const onAccountChange = (event: React.SyntheticEvent<HTMLElement, Event>, data: DropdownProps) => {
+		const addressValue = data.value as string;
+		setAddress(addressValue);
+	};
+
+	const handleSetCredentials = async (data: Record<string, any>): Promise<void> => {
+		if (!accounts.length) {
+			return getAccounts();
+		}
+
+		const { username, password } = data;
+
+		try {
+			const injected = await web3FromSource(accounts[0].meta.source);
+			const signRaw = injected && injected.signer && injected.signer.signRaw;
+
+			if (!signRaw) {
+				return console.error('Signer not available');
+			}
+
+			const { data: startResult } = await setCredentialsStartMutation({
+				variables: {
+					address
+				}
+			});
+
+			const signMessage = startResult?.setCredentialsStart?.signMessage;
+
+			if (!signMessage) {
+				throw new Error('Challenge message not found');
+			}
+
+			const { signature } = await signRaw({
+				address,
+				data: stringToHex(signMessage),
+				type: 'bytes'
+			});
+
+			const { data: setResult } = await setCredentialsConfirmMutation({
+				variables: {
+					address,
+					password,
+					signature,
+					username
+				}
+			});
+
+			if (setResult?.setCredentialsConfirm?.token) {
+				handleTokenChange(setResult?.setCredentialsConfirm?.token);
+			}
+			currentUser.setUserDetailsContextState((prevState) => {
+				return {
+					...prevState,
+					username,
+					web3signup: false
+				};
+			});
+		} catch (error) {
+			setErr(error);
+		}
+	};
+
+	const dismissModal = () => {
+		setShowModal(false);
+	};
+
+	const openModal = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>):void => {
+		event.preventDefault();
+		event.stopPropagation();
+
+		setShowModal(true);
+	};
+
+	return (
+		<>
+			{showModal ?
+				<Modal
+					buttons={
+						<Form.Group>
+							<Button
+								content='Set Username Password'
+								disabled={loading}
+								primary
+								onClick={handleSubmit(handleSetCredentials)}
+							/>
+							<Button
+								content='Close'
+								secondary
+								onClick={dismissModal}
+							/>
+						</Form.Group>
+					}
+					centered
+					dimmer='inverted'
+					open
+					onClose={dismissModal}
+					size='tiny'
+					title={'Set Username password'}
+				>
+					<Form standalone={false} onSubmit={handleSubmit(handleSetCredentials)}>
+						{extensionNotFound?
+							<div className='card'>
+								<ExtensionNotDetected />
+							</div>
+							: null
+						}
+						{accountsNotFound?
+							<div className='card'>
+								<div className='text-muted'>You need at least one account in Polkadot-js extenstion to login.</div>
+								<div className='text-muted'>Please reload this page after adding accounts.</div>
+							</div>
+							: null
+						}
+						{isAccountLoading
+							?
+							<Loader text={'Requesting Web3 accounts'}/>
+							:
+							accounts.length > 0 &&
+							<>
+								<Form.Group>
+									<AccountSelectionForm
```

Comment:
I don't think we need an account selection, using the default account is good enough.

Reply:
removed